apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.cs

```diff
@@ -178,46 +178,51 @@ public DbType DbType
+        /// <returns>The configured <see cref="System.Type"/>.</returns>
+        internal System.Type GetConfiguredClrType(SpannerConversionOptions options)
         {
-            get
+            switch (TypeCode)
             {
-                switch (TypeCode)
-                {
-                    case TypeCode.Bool:
-                        return typeof(bool);
-                    case TypeCode.Int64:
-                        return typeof(long);
-                    case TypeCode.Float64:
-                        return typeof(double);
-                    case TypeCode.Timestamp:
-                    case TypeCode.Date:
-                        return typeof(DateTime);
-                    case TypeCode.String:
-                        return typeof(string);
-                    case TypeCode.Bytes:
-                        return typeof(byte[]);
-                    case TypeCode.Array:
-                        return typeof(List<>).MakeGenericType(ArrayElementType.DefaultClrType);
-                    case TypeCode.Struct:
-                        return typeof(SpannerStruct);
-                    case TypeCode.Numeric:
-                        if (TypeAnnotationCode == TypeAnnotationCode.PgNumeric)
-                        {
-                            return typeof(PgNumeric);
-                        }
-                        return typeof(SpannerNumeric);
-                    case TypeCode.Json:
-                        return typeof(string);
-                    default:
-                        //if we don't recognize it (or its a struct), we use the google native wellknown type.
-                        return typeof(Value);
-                }
+                case TypeCode.Bool:
+                    return typeof(bool);
+                case TypeCode.Int64:
+                    return typeof(long);
+                case TypeCode.Float64:
+                    return typeof(double);
+                case TypeCode.Timestamp:
+                case TypeCode.Date:
+                    return typeof(DateTime);
+                case TypeCode.String:
+                    return typeof(string);
+                case TypeCode.Bytes:
+                    return typeof(byte[]);
+                case TypeCode.Array:
+                    return typeof(List<>).MakeGenericType(ArrayElementType.GetConfiguredClrType(options));
+                case TypeCode.Struct:
+                    return typeof(SpannerStruct);
+                case TypeCode.Numeric:
+                    if (TypeAnnotationCode == TypeAnnotationCode.PgNumeric)
+                    {
+                        return typeof(PgNumeric);
+                    }
+                    return typeof(SpannerNumeric);
+                case TypeCode.Json:
+                    return typeof(string);
+                default:
+                    //if we don't recognize it (or its a struct), we use the google native wellknown type.
```

Comment:
Let's fix this comment while we're here:

```suggestion
                    // If we don't recognize it (or it's a struct), we use the protobuf Value well-known type.
```